api/app/services/rag/document_retriever.py

```
@@ -455,12 +538,87 @@ def retrieve_with_scores(
+        def _absolute_scores(
+            k: int, filter_dict: dict[str, Any] | None
+        ) -> dict[tuple[str, str], float]:
+            semantic_retrieve = getattr(
+                self.retriever, "retrieve_semantic_with_scores", None
+            )
+            if not callable(semantic_retrieve):
+                return {}
+
+            try:
+                results = semantic_retrieve(query, k=k, filter_dict=filter_dict)
+            except Exception:
```

Ruff (0.15.20)

[warning] Do not catch blind exception: `Exception`

(BLE001)